test/integration/integrationTestUtils/waiters.js

```diff
@@ -0,0 +1,63 @@
+const { assert } = require("chai");
+const {ContinueLoopError, WaitIsOverError} = require("./customeErrors");
+require('log-timestamp');
+
+
+const WAIT=300
+const MAX_REPETITIONS= 50
+const MAX_WAIT_TIME = 10000
+
+const sleep = (time) => {
+    return new Promise((resolve) => setTimeout(resolve, time));
+};
+
+/**
+ * Wait until function returns results or exit in time had passed or exception was thrown.
+ * @param func {function} Run this function over and over
+ * @param wait {number} Initial time wait. Pay attention that exponential backoff is used here, e.g. wait * 2 for each repetition until maxRepetitions have reached.
+ * @param givenMaxRepetitions {number} How many repetition to run
+ * @param givenMaxWaitTime {number} Maximum wait time
+ * @returns {Promise<*>}
+ */
+async function syncLoop(func, wait, givenMaxRepetitions, givenMaxWaitTime = MAX_WAIT_TIME) {
```

Comment:
Do we still need it with the `waitOn` package?

Reply:
yes, we use it to wait for spans in the file